server/index.js

```diff
@@ -52,6 +51,86 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+
+  if (!req.session.cartId) {
+    res.status(200).json([]);
+  } else {
+    const sql = `select "c"."cartItemId",
+    "c"."price",
+    "p"."productId",
+    "p"."image",
+    "p"."name",
+    "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using("productId")
+    where "c"."cartId" = $1`;
+    const cartId = req.session.cartId;
+    db.query(sql, [cartId])
+      .then(shoppingCart => {
+        return Object.assign(shoppingCart, cartId);
+      });
+  }
+});
+
+app.post('/api/cart', (req, res) => {
+  const { productId } = req.body;
+  if (!(Number(productId)) || productId <= 0) {
+    res.status(400).json({ err: 'ProductID should be positive number' });
+  }
+  const sql = `
+  select "price"
+  from "products"
+  where "productId" = $1
+  `;
+  const body = [productId];
+  db.query(sql, body)
+    .then(price => {
+      if (req.session.cartId) {
+        const cookie = { cartId: req.session.cartId, price: price.rows[0].price };
+        return cookie;
+      } else {
+        const sql = `insert into "carts"("cartId", "createdAt")
+        values(default, default )
+        returning "cartId"`;
+        return db.query(sql)
+          .then(cartId => {
+            return Object.assign(price.rows[0], cartId.rows[0]);
+          });
+      }
+    })
+    .then(combinedObj => {
+      req.session.cartId = combinedObj.cartId;
+      const sql = `insert into "cartItems" ("cartId", "productId", "price")
+    values ($1, $2, $3)
+    returning "cartItemId"`;
+      return db.query(sql, [combinedObj.cartId, productId, combinedObj.price])
+        .then(cartItemId => {
+          return Object.assign(cartItemId, combinedObj.cartId, productId, combinedObj.price);
+        });
+    })
+    .then(cartItems => {
+      const sql = `
+      select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+      where "c"."cartItemId" = $1`;
+      const cartItemId = cartItems.rows[0].cartItemId;
+      return db.query(sql, [cartItemId])
+        .then(fields => {
+          return Object.assign(fields, cartItemId);
+        })
+        .then(fieldsObj => {
+          res.status(201).json({ success: 'Cart Item Works!' });
+        });
+    });
```

Comment:
Your last `.then()` should have a `.catch()` chained on to it so you can handle the error if your promise chain is rejected.